source/device_defender.c

```diff
@@ -502,9 +754,16 @@ struct aws_iotdevice_defender_v1_task *aws_iotdevice_defender_v1_report_task(
     defender_task->previous_net_xfer.packets_in = 0;
     defender_task->previous_net_xfer.packets_out = 0;
     defender_task->has_previous_net_xfer = false;
-    defender_task->config = *config;
+    memcpy(&defender_task->config, config, sizeof(struct aws_iotdevice_defender_report_task_config));
```

Comment:
It's awkward but there are a lot of instances where we have a config (where everything is by value, shallow) and a persistent internal structure that owns all of its resources.  I don't think we should be memcpying configs that contain pointers, even if it's "safe" from a lifecycle standpoint.

Reply:
Definitely something that was meant to be adjusted back. Changing to regular assignment works fine here.